app/src/main/java/org/mozilla/fenix/home/recenttabs/view/RecentTabs.kt

```diff
@@ -224,48 +229,117 @@ private fun RecentSearchGroupItem(
-                            true -> Color(0xFF42414D) // DarkGrey30
-                            false -> PhotonColors.LightGrey30
-                        }
-                    )
+    Card(
+        modifier = modifier
+    ) {
+        when {
+            thumbnail != null -> {
+                Image(
+                    bitmap = thumbnail.asImageBitmap(),
+                    contentDescription = null,
+                    modifier = modifier,
+                    contentScale = ContentScale.FillWidth,
+                    alignment = Alignment.TopCenter
                 )
             }
-
-            WithIcon { icon ->
+            icon != null -> {
                 Image(
-                    painter = icon.painter,
+                    bitmap = icon.asImageBitmap(),
                     contentDescription = null,
-                    modifier = modifier
+                    modifier = iconModifier,
+                    contentScale = ContentScale.Fit,
+                    alignment = Alignment.Center
                 )
             }
+            else -> {
+                components.core.icons.Loader(
+                    url = url
+                ) {
+                    Placeholder {
+                        Box(
+                            modifier = Modifier.background(
+                                color = when (isSystemInDarkTheme()) {
+                                    true -> Color(0xFF42414D) // DarkGrey30
+                                    false -> PhotonColors.LightGrey30
+                                }
+                            )
+                        )
+                    }
+
+                    WithIcon { icon ->
+                        Image(
+                            painter = icon.painter,
+                            contentDescription = null,
+                            modifier = iconModifier,
+                            contentScale = ContentScale.Fit
+                        )
+                    }
+                }
+            }
+        }
+    }
+}
+
+/**
+ * A recent tab icon.
+ *
+ * @param url The loaded URL of the tab.
+ * @param modifier modifier Modifier used to draw the image content.
+ * @param icon The icon of the tab.
+ * is null.
+ */
+@Composable
+private fun RecentTabIcon(
+    url: String,
+    modifier: Modifier = Modifier,
+    icon: Bitmap? = null,
+) {
+    when {
+        icon != null -> {
+            Image(
+                bitmap = icon.asImageBitmap(),
+                contentDescription = null,
+                modifier = modifier
+            )
+        }
+        else -> {
+            components.core.icons.Loader(
+                url = url
+            ) {
+                Placeholder {
+                    Box(
+                        modifier = Modifier.background(
+                            color = when (isSystemInDarkTheme()) {
+                                true -> Color(0xFF42414D) // DarkGrey30
```

Comment:
How come an explicit color hex here?

Reply:
Good question.  I don't know why.  I got this from top sites.  I'll look into this.  Thanks,

Reply:
At that point, https://github.com/mozilla-mobile/android-components/pull/10954 had not landed yet, so we kept a reference to what the colour name should be when we were able to use it.